lib/whitespace.coffee

```diff
@@ -87,3 +97,11 @@ class Whitespace
 
     buffer.transact ->
       buffer.scan new RegExp(spacesText, 'g'), ({replace}) -> replace('\t')
+
+  clearBlankLineOnNewline: (editor, event) ->
+    return unless event.text is "\n"
+    buffer = editor.getBuffer()
+    previousRow = editor.getCursorBufferPosition().row - 1
```

Comment:
I planned on using it but when I went to look at the documentation for [editor.onDidInsert](https://github.com/atom/atom/blob/master/src/text-editor.coffee#L334) it didn't mention a range in the event object.